backend/controller/timeline/timeline.go

```diff
@@ -2,124 +2,55 @@ package timeline
-	"github.com/TBD54566975/ftl/backend/schema"
-	"github.com/TBD54566975/ftl/internal/log"
-	"github.com/TBD54566975/ftl/internal/model"
 )
 
-type Service struct {
-	dal *dal.DAL
-}
-
-func New(ctx context.Context, conn libdal.Connection, encryption *encryption.Service) *Service {
-	return &Service{dal: dal.New(conn, encryption)}
-}
-
-type Log struct {
-	DeploymentKey model.DeploymentKey
-	RequestKey    optional.Option[model.RequestKey]
-	Msg           *ftlv1.StreamDeploymentLogsRequest
-}
-
-func (s *Service) RecordLog(ctx context.Context, log *Log) error {
-	err := s.dal.InsertLogEvent(ctx, &dal.LogEvent{
-		RequestKey:    log.RequestKey,
-		DeploymentKey: log.DeploymentKey,
-		Time:          log.Msg.TimeStamp.AsTime(),
-		Level:         log.Msg.LogLevel,
-		Attributes:    log.Msg.Attributes,
-		Message:       log.Msg.Message,
-		Error:         optional.Ptr(log.Msg.Error),
-	})
-	if err != nil {
-		return fmt.Errorf("failed to insert log event: %w", err)
-	}
-	return nil
-}
+type EventType = sql.EventType
 
-type Call struct {
-	DeploymentKey    model.DeploymentKey
-	RequestKey       model.RequestKey
-	ParentRequestKey optional.Option[model.RequestKey]
-	StartTime        time.Time
-	DestVerb         *schema.Ref
-	Callers          []*schema.Ref
-	Request          *ftlv1.CallRequest
-	Response         either.Either[*ftlv1.CallResponse, error]
-}
-
-func (s *Service) RecordCall(ctx context.Context, call *Call) {
-	logger := log.FromContext(ctx)
-	var sourceVerb optional.Option[schema.Ref]
-	if len(call.Callers) > 0 {
-		sourceVerb = optional.Some(*call.Callers[0])
-	}
-
-	var errorStr optional.Option[string]
-	var stack optional.Option[string]
-	var responseBody []byte
-
-	switch response := call.Response.(type) {
-	case either.Left[*ftlv1.CallResponse, error]:
-		resp := response.Get()
-		responseBody = resp.GetBody()
-		if callError := resp.GetError(); callError != nil {
-			errorStr = optional.Some(callError.Message)
-			stack = optional.Ptr(callError.Stack)
-		}
-	case either.Right[*ftlv1.CallResponse, error]:
-		callError := response.Get()
-		errorStr = optional.Some(callError.Error())
-	}
+// Supported event types.
+const (
+	EventTypeLog               = sql.EventTypeLog
+	EventTypeCall              = sql.EventTypeCall
+	EventTypeDeploymentCreated = sql.EventTypeDeploymentCreated
+	EventTypeDeploymentUpdated = sql.EventTypeDeploymentUpdated
+)
 
-	err := s.dal.InsertCallEvent(ctx, &dal.CallEvent{
-		Time:             call.StartTime,
-		DeploymentKey:    call.DeploymentKey,
-		RequestKey:       optional.Some(call.RequestKey),
-		ParentRequestKey: call.ParentRequestKey,
-		Duration:         time.Since(call.StartTime),
-		SourceVerb:       sourceVerb,
-		DestVerb:         *call.DestVerb,
-		Request:          call.Request.GetBody(),
-		Response:         responseBody,
-		Error:            errorStr,
-		Stack:            stack,
-	})
-	if err != nil {
-		logger.Errorf(err, "failed to record call")
-	}
+// TimelineEvent types.
+//
+//sumtype:decl
```

Comment:
Does this do anything outside of a module? :o

Reply:
I think it's for the sumtype tool in go to make this interface as a sumtype.

Reply:
ahhhh got it